identity/tests/db_queries.rs

```diff
@@ -0,0 +1,336 @@
+        .await
+        .expect("Could not set up test environment");
+
+    let expected_result = vec![
+        User {
+            id: 3,
+            sub: "3".into(),
+            email: "justin.wilkins@example.net".into(),
+            given_name: "Justin".into(),
+            family_name: "Wilkins".into(),
+            picture: "https://example.com/avatar.jpg".into(),
+            oauth_access_token: "access_token".into(),
+            oauth_access_token_valid: NaiveDate::from_ymd(2020, 12, 31).and_hms(0, 0, 0),
+            oauth_refresh_token: "refresh_token".into(),
+            active: true,
+        },
+        User {
+            id: 5,
+            sub: "5".into(),
+            email: "richard.henderson@example.net".into(),
+            given_name: "Richard".into(),
+            family_name: "Henderson".into(),
+            picture: "https://example.com/avatar.jpg".into(),
+            oauth_access_token: "access_token".into(),
+            oauth_access_token_valid: NaiveDate::from_ymd(2020, 12, 31).and_hms(0, 0, 0),
+            oauth_refresh_token: "refresh_token".into(),
+            active: true,
+        },
+    ];
+
+    // Act
+    let result = queries::list_users(2, 2, Some(true), &db_pool.get().unwrap());
+
+    // Assert
+    assert_eq!(Ok(expected_result), result);
+}
+
+// list inactive users with offset/limit
+#[tokio::test]
+async fn db_list_users_exists_inactive() {
+    // Arrange
+    let (_configuration, db_pool, _db_test_context) = setup(stdext::function_name!().into())
+        .await
+        .expect("Could not set up test environment");
+
+    let expected_result = vec![User {
+        id: 4,
+        sub: "4".into(),
+        email: "tim.jackson@example.net".into(),
+        given_name: "Tim".into(),
+        family_name: "Jackson".into(),
+        picture: "https://example.com/avatar.jpg".into(),
+        oauth_access_token: "access_token".into(),
+        oauth_access_token_valid: NaiveDate::from_ymd(2020, 12, 31).and_hms(0, 0, 0),
+        oauth_refresh_token: "refresh_token".into(),
+        active: false,
+    }];
+
+    // Act
+    let result = queries::list_users(0, 1, Some(false), &db_pool.get().unwrap());
+
+    // Assert
+    assert_eq!(Ok(expected_result), result);
+}
+
+// verify update user
+#[tokio::test]
+async fn db_update_user_verify() {
+    // Arrange
+    let (_configuration, db_pool, _db_test_context) = setup(stdext::function_name!().into())
+        .await
+        .expect("Could not set up test environment");
+
+    let expected_result = User {
+        id: 2,
+        sub: "2".into(),
+        email: "jack.kerr@example.net".into(),
+        given_name: "Jack".into(),
+        family_name: "Kerr".into(),
+        picture: "https://example.com/avatar.jpg".into(),
+        oauth_access_token: "access_token".into(),
+        oauth_access_token_valid: NaiveDate::from_ymd(2020, 12, 31).and_hms(0, 0, 0),
+        oauth_refresh_token: "refresh_token".into(),
+        active: false,
+    };
```

Comment:
Maybe move that into a separate module since you need it frequently?

Reply:
I only use this specific one twice. The refactoring will change this models anyway so I'm not too worried about it.